match/lib/match/encryption/interface.rb

```diff
@@ -0,0 +1,29 @@
+module Match
+  module Encryption
+    class Interface
+      # Returns the class to be used for a given `storage_mode`
+      def self.encryption_class_for_storage_mode(storage_mode)
```

Comment:
This interface seems a bit clunky, did you have some usage in mind that does not always look like this?

```
self.storage = Storage::Interface.storage_class_for_storage_mode(params[:storage_mode]).new
```

Otherwise this could be more elegantly replaced by adding a helper function to the module to something like 

```
self.storage = Storage.for_mode(params[:storage_mode])
```

Reply:
Ohhhhh, I think I like that too... I will make that change 😊